core/src/test/scala/akka/persistence/r2dbc/query/EventsBySlicePerfSpec.scala

```diff
@@ -83,22 +106,98 @@ class EventsBySlicePerfSpec
         val counts: Seq[Future[Int]] = ranges.map { range =>
           query
             .currentEventsBySlices[String](entityType, range.min, range.max, NoOffset)
-            .runWith(Sink.fold(0) { case (acc, _) =>
-              if (acc > 0 && acc % 100 == 0)
-                println(s"#$iteration Reading [$acc] events from slices [${range.min}-${range.max}] " +
-                s"took [${(System.nanoTime() - t1) / 1000 / 1000}] ms")
-              acc + 1
+            .runWith(Sink.fold(0) { case (acc, env) =>
+              if (EnvelopeOrigin.fromQuery(env)) {
+                if (acc > 0 && acc % 100 == 0)
+                  println(s"#$iteration Reading [$acc] events from slices [${range.min}-${range.max}] " +
+                  s"took [${(System.nanoTime() - t1) / 1000 / 1000}] ms")
+                acc + 1
+              } else {
+                acc
+              }
             })
         }
         implicit val ec: ExecutionContext = testKit.system.executionContext
         val total = Await.result(Future.sequence(counts).map(_.sum), 30.seconds)
         total shouldBe totalNumberOfEvents
         println(
-          s"#$iteration Reading all [$totalNumberOfEvents] events from [${ranges.size}] eventsBySlices " +
+          s"#$iteration Reading all [$totalNumberOfEvents] events from [${ranges.size}] slices with currentEventsBySlices " +
           s"took [${(System.nanoTime() - t1) / 1000 / 1000}] ms")
       }
     }
 
+    "write and read concurrently" in {
```

Comment:
Used this new test and looked at logs to verify that the queries were executed as expected.